src/lib/exponent/index.ts

```diff
@@ -0,0 +1,103 @@
+import {z} from 'zod';
+
+import {ERRORS} from '../../util/errors';
+import {buildErrorMessage} from '../../util/helpers';
+import {validate} from '../../util/validations';
+
+import {PluginInterface} from '../../interfaces';
+import {PluginParams} from '../../types/common';
+import {ExponentConfig} from './types';
+
+const {InputValidationError} = ERRORS;
+
+export const Exponent = (globalConfig: ExponentConfig): PluginInterface => {
+  const errorBuilder = buildErrorMessage(Exponent.name);
+  const metadata = {
+    kind: 'execute',
+  };
+
+  /**
+   * Checks global config value are valid.
+   */
+  const validateGlobalConfig = () => {
+    const globalConfigSchema = z.object({
+      'input-parameter': z.string().min(1),
+      exponent: z.number().min(1),
+      'output-parameter': z.string().min(1),
+    });
+
+    return validate<z.infer<typeof globalConfigSchema>>(
+      globalConfigSchema,
+      globalConfig
+    );
+  };
+
+  /**
+   * Checks for required fields in input.
+   */
+  const validateSingleInput = (input: PluginParams, inputParameter: string) => {
+    validateParamExists(input, inputParameter);
+    validateNumericString(input[inputParameter]);
+    return input;
+  };
+
+  const validateParamExists = (input: PluginParams, param: string) => {
+    if (!input[param]) {
+      throw new InputValidationError(
+        errorBuilder({
+          message: `${param} is missing from the input array`,
+        })
+      );
+    }
+  };
+
+  const validateNumericString = (str: string) => {
+    if (isNaN(+Number(str))) {
```

Comment:
the condition will be more readable with this `typeof str !== 'number'`

Reply:
cool. also added a unit test to cover this scenario.
I have to say this was a bit counter intuitive for me: if we pass str as `str: string`, how come `typeof str !== 'number'` isn't always true? guess it's one of those unique TypeScript things...

Reply:
I don't think this change works.

`typeof str !== 'number'` does not check that the string is numeric, it checks whether the type is a number.

Depending on the purpose of the check, you might want to check that the string *parses* to a number, or use some regex to check that the string only contains numbers.

@manushak please confirm

Reply:
you're right @jmcook1186 , but shouldn't we require the user to pass a number instead of a string, like this: '10'?

Reply:
@jmcook1186 your turn :)

Reply:
Hi @pazbardanl @manushak we should be able to parse numeric strings too. Can you confirm that this is the current behaviour.

Reply:
In this case @pazbardanl, please, use your version

Reply:
done